packages/@aws-cdk/aws-apigatewayv2-authorizers/README.md

```diff
@@ -192,3 +193,25 @@ api.addRoutes({
   authorizer,
 });
 ```
+
+## IAM Authorizers
```

Comment:
Hi @iRoachie -

This does not feel like the right approach to solving IAM authorizers. Just having an `HttpIamAuthorizer` class to set the `AuthorizationType` and then leaving it up to the user to correctly configure the IAM policy feels sub-optimal.

I would suggest going with an alternate approach. Here's my proposal written in the form of what the user experience will be -

```ts
const api = new HttpApi(stack, 'HttpApi');
const routes: Route[] = api.addRoutes({ ... });
routes[0].grantInvoke(principal, { // principal is of type iam.IPrincipal
  httpMethod: 'GET'
});
```

This should, under the hood, automatically set the `AuthorizationType` to `AWS_IAM` and attach the relevant IAM permissions to the principal.

Reply:
I was going based on usage and what was requested in #10534.

As far as I understand the use case for IAM Authorization is to set policies on individual users to grant access to all/some routes.

I see where you are going with your suggestion however I think this can be an add-on to what is currently in this PR. If we only change the authorizationType when grantInvoke is called on a route then we automatically rule out scenarios where user permissions are managed outside of cdk/cloudformation (which is pretty common for physical users).

So just to be clear they're two things here:
1. Setting route to use IAM Authorization
2. Allowing an easy way to assign the relevant policies to a user/principal so that it can invoke that route.